### PR TITLE
Update RxSwift dependency to 4.0.0-beta.0

### LINKED ISSUE #1303: Swift 4 and RxMoya
Hey guys,

RxMoya currently doesn't support Swift 4 because there was no RxSwift release for Swift 4. However, RxSwift recently released [4.0.0 beta](https://github.com/ReactiveX/RxSwift/releases/tag/4.0.0-beta.0) and I think we can start using this version to support Swift 4.0 in RxMoya. How do you think?

### PR DESCRIPTION
**Pull request by devxoul**
Resolves #1303

### CODE REVIEW

**Comment by devxoul** [comment]
Is there any reason that Moya doesn't have `.swift-version` file?

File: Moya.podspec [hunk 1]
```diff
@@ -40,6 +40,6 @@ Pod::Spec.new do |s|
   s.subspec "RxSwift" do |ss|
     ss.source_files = "Sources/RxMoya/"
     ss.dependency "Moya/Core"
-    ss.dependency "RxSwift", "~> 3.3"
+    ss.dependency "RxSwift", "=> 4.0.0-beta.0"
```
Comment (pedrovereza) [inline Moya.podspec]: I think this should be `~> 4.0`

**Comment by pedrovereza** [comment]
Also, should we change the target of this PR to `10.0.0-dev` as RxSwift 4 is not stable yet?

File: Moya.podspec (hunk 1 above)
Comment (devxoul) [reply Moya.podspec]: I thought that `~> 4.0` doesn't contain if there's a `-alpha` or `-beta`. Let me check it again with a new project :)

Comment (devxoul) [reply Moya.podspec]: **Podfile**

```ruby
pod 'RxSwift', '4.0.0-beta.0'
pod 'Moya', :git => 'https://github.com/devxoul/Moya.git', :branch => 'rxswift4-version'
pod 'Moya/RxSwift', :git => 'https://github.com/devxoul/Moya.git', :branch => 'rxswift4-version'
```

**Result**

```
[!] Unable to satisfy the following requirements:

- `RxSwift (= 4.0.0-beta.0)` required by `Podfile`
- `RxSwift (~> 4.0)` required by `Moya/RxSwift (9.0.0)`
```

**Comment by devxoul** [comment]
@pedrovereza, not sure. Please change the base branch before being merged if needed :)

File: Moya.podspec (hunk 1 above)
Comment (pedrovereza) [reply Moya.podspec]: Fair enough. I also see that you fixed the operator from `=>` to `>=` 👍

Comment (devxoul) [reply Moya.podspec]: Yeah, I'm always confused with `=>` and `>=` 😅

**Comment by devxoul** [comment]
@pedrovereza Updated Package.swift :)

**Comment by BasThomas** [comment]
On the branch: we should, so I changed it to the dev branch 👍

**Comment by devxoul** [comment]
@BasThomas, thanks! I just rebased `10.0.0-dev` branch and force-pushed.

**Comment by SD10** [comment]
Build Error:

Resolving dependencies of `Podfile`
[!] Due to the previous naïve CocoaPods resolver, you were using a pre-release version of `RxSwift`, without explicitly asking for a pre-release version, which now leads to a conflict. 

Please decide to either use that pre-release version by adding the version requirement to your Podfile (e.g. `pod 'RxSwift', '= 4.0.0-alpha.1'`) or revert to a stable version by running `pod update RxSwift`.

**Comment by pedrovereza** [comment]
Seems like the [`Podfile.lock` on the Demo project](https://github.com/Moya/Moya/blob/master/Demo/Podfile.lock) needs to be updated

**Comment by volodg** [comment]
@pedrovereza Can be it approved now?

**Comment by sunshinejr** [comment]
Because I had to rebase 10.0.0-dev, I resolved conflicts here as well. But then I figured that we probably don't want this merge and rebase would be more preferable. Would you be able to remove my merge commit and rebase this one, @devxoul? Sorry for the trouble.

**Comment by sunshinejr** [comment]
Thanks again, @devxoul! 🎉

**Pull request merged**